mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp

```diff
@@ -367,12 +367,13 @@ class AccessChainPattern : public SPIRVToLLVMConversion<spirv::AccessChainOp> {
     Value zero = rewriter.create<LLVM::ConstantOp>(
         op.getLoc(), llvmIndexType, rewriter.getIntegerAttr(indexType, 0));
     indices.insert(indices.begin(), zero);
-    rewriter.replaceOpWithNewOp<LLVM::GEPOp>(
-        op, dstType,
-        typeConverter.convertType(
-            cast<spirv::PointerType>(op.getBasePtr().getType())
-                .getPointeeType()),
-        adaptor.getBasePtr(), indices);
+
+    auto elementType = typeConverter.convertType(
+        cast<spirv::PointerType>(op.getBasePtr().getType()).getPointeeType());
+    if (!elementType)
+      return failure();
```

Comment:
```suggestion
      return rewriter.notifyMatchFailure(op, "type conversion failed");
```

Reply:
Thanks! I've noted the purpose of `notifyMatchFailure` for failure logging in debug mode. Should we use `notifyMatchFailure` to handle `convertType` failures elsewhere in the code, such L389, L407?

Reply:
Yeah, that would be nice to have

Reply:
I followed it elsewhere at https://github.com/llvm/llvm-project/pull/79955/commits/ea81352d770087f789d610fc5ad1f45d7147def5

Reply:
Thanks @sott0n! This is very welcome but in the future it would be better to have similar improvements in a separate PR, so that it's easier to review bugfixes/new features without the noise of changes not related to that per se. In this instance it fine, I'd keep it as-is since you've already put everything together.

Reply:
Got it, thanks for your clarification.